### PR TITLE
Refactor source

### PR DESCRIPTION
**Pull request by CytraL**
This PR try to clean the source... is only for read better and improve a bit in time and space.
- Not adds any new feature.
- Not removes any old feature.

PR not tested... only tried that compile but not if i broke something.
Not merge... :)

Please comment if this is necessary... perhaps not like this on the source or like do something more big and organized... ?¿??

### CODE REVIEW

File: src/game/editor/io.cpp [hunk 1]
```diff
@@ -269,9 +269,10 @@ int CEditorMap::Save(class IStorage *pStorage, const char *pFileName)
 
 			char *pSettings = (char *)mem_alloc(Size, 1);
 			char *pNext = pSettings;
+			int Length = 0;
 			for(int i = 0; i < m_lSettings.size(); i++)
 			{
-				int Length = str_length(m_lSettings[i].m_aCommand) + 1;
+				Length = str_length(m_lSettings[i].m_aCommand) + 1;
```
Comment (tbu-) [inline src/game/editor/io.cpp]: This change doesn't seem useful.

**Comment by CytraL** [comment]
Description updated.

**Comment by CytraL** [comment]
No comments = best do other stuff?

**Comment by def-** [comment]
I don't have much time for DDNet. Maybe you can motivate someone else to review.

**Comment by heinrich5991** [comment]
Generally looks good.

Why do you lift some of the variables out of the loops? I think that's bad practice.

Also, the hardcoded lengths of the arrays isn't that nice, you can get the length of the array by `sizeof(array)/sizeof(array[0])`

**Comment by CytraL** [comment]
Yeah, perhaps for non-object isn't good practice... but principally for this:
https://google.github.io/styleguide/cppguide.html#Local_Variables

And yes again... i will update the source with the suggested changes.

**Comment by heinrich5991** [comment]
From your link:

>Variables needed for if, while and for statements should normally be declared within those statements, so that such variables are confined to those scopes.
>There is one caveat: if the variable is an object, its constructor is invoked every time it enters scope and is created, and its destructor is invoked every time it goes out of scope.

This means it should generally be declared inside the loop, *unless* there's a constructor/destructor that gets called after every loop iteration.

**Comment by heinrich5991** [comment]
Thanks for your contribution, I've completed the refactor in #671 and will now close this PR.

**Pull request closed without merging**